SecurityPkg/Library/SecureBootVariableLib/SecureBootVariableLib.c

```diff
@@ -578,9 +578,12 @@ DeletePlatformKey (
 {
   EFI_STATUS  Status;
 
-  Status = SetSecureBootMode (CUSTOM_SECURE_BOOT_MODE);
-  if (EFI_ERROR (Status)) {
-    return Status;
+  // MU_CHANGE: Remove the indicator of custom secure boot mode variable
+  if (FeaturePcdGet (PcdSupportCustomSecureBootModeVariable)) {
+    Status = SetSecureBootMode (CUSTOM_SECURE_BOOT_MODE);

```

Comment:
So just to explain what I've learned about this,

CUSTOM_SECURE_BOOT_MODE / STANDARD_SECURE_BOOT_MODE (i.e "CustomMode") is not a concept defined in the UEFI Specification. STANDARD_SECURE_BOOT_MODE  is the implicit default mode as per the UEFI Specification. 

"CustomMode" is defined in the [EDKII implementation](https://cdrdv2-public.intel.com/671120/a-tour-beyond-bios-implementing-uefi-authenticated-variables-in-smm-with-edkii.pdf) and required by the [windows hardware certification requirements for client and server systems](https://learn.microsoft.com/en-us/previous-versions/windows/hardware/cert-program/windows-hardware-certification-requirements-for-client-and-server-systems?redirectedfrom=MSDN#:~:text=On%20an%20ARM%20system%2C%20it%20is%20forbidden%20to%20enable%20Custom%20Mode.%20Only%20Standard%20Mode%20may%20be%20enabled) and described by the [Signing UEFI Applications And Drivers for UEFI Secure Boot](https://github.com/tianocore-docs/Docs/raw/master/User_Docs/SigningUefiImages%20-v1dot31.pdf) documentation.

Quote:
"On non-ARM systems, the platform MUST implement the ability for a physically present user to select between two Secure Boot modes in firmware setup: "Custom" and "Standard". Custom Mode allows for more flexibility as specified in the following: 
[...]

The firmware setup shall indicate if Secure Boot is turned on, and if it is operated in Standard or Custom Mode. The firmware setup must provide an option to return from Custom to Standard Mode which restores the factory defaults. On an ARM system, it is forbidden to enable Custom Mode. Only Standard Mode may be enabled
"

So as long as this change, on any arm based system, leaves the device in STANDARD_SECURE_BOOT_MODE and any non-arm based system has this Pcd. Then this change does not conflict with any specification or requirement.